runtime/android/core/src/org/xwalk/core/XWalkFileChooser.java

```diff
@@ -0,0 +1,288 @@
+            if (capture.equals("true")) {
+                if (acceptType.startsWith(IMAGE_TYPE)) {
+                    if (takePictureIntent != null) {
+                        mActivity.startActivityForResult(takePictureIntent, INPUT_FILE_REQUEST_CODE);
+                        Log.d(TAG, "Started taking picture");
+                        return true;
+                    }
+                } else if (acceptType.startsWith(VIDEO_TYPE)) {
+                    mActivity.startActivityForResult(camcorder, INPUT_FILE_REQUEST_CODE);
+                    Log.d(TAG, "Started camcorder");
+                    return true;
+                } else if (acceptType.startsWith(AUDIO_TYPE)) {
+                    mActivity.startActivityForResult(soundRecorder, INPUT_FILE_REQUEST_CODE);
+                    Log.d(TAG, "Started sound recorder");
+                    return true;
+                }
+            } else {
+                if (acceptType.startsWith(IMAGE_TYPE)) {
+                    if (takePictureIntent != null) {
+                        extraIntents.add(takePictureIntent);
+                    }
+                    contentSelectionIntent.setType(ALL_IMAGE_TYPES);
+                } else if (acceptType.startsWith(VIDEO_TYPE)) {
+                    extraIntents.add(camcorder);
+                    contentSelectionIntent.setType(ALL_VIDEO_TYPES);
+                } else if (acceptType.startsWith(AUDIO_TYPE)) {
+                    extraIntents.add(soundRecorder);
+                    contentSelectionIntent.setType(ALL_AUDIO_TYPES);
+                }
+            }
+        }
+
+        // Couldn't resolve an accept type.
+        if (extraIntents.isEmpty() && canWriteExternalStorage()) {
+            if (takePictureIntent != null) {
+                extraIntents.add(takePictureIntent);
+            }
+            extraIntents.add(camcorder);
+            extraIntents.add(soundRecorder);
+            contentSelectionIntent.setType(ANY_TYPES);
+        }
+
+        Intent chooserIntent = new Intent(Intent.ACTION_CHOOSER);
+        chooserIntent.putExtra(Intent.EXTRA_INTENT, contentSelectionIntent);
+        if (!extraIntents.isEmpty()) {
+            chooserIntent.putExtra(Intent.EXTRA_INITIAL_INTENTS,
+                    extraIntents.toArray(new Intent[] { }));
+        }
+        mActivity.startActivityForResult(chooserIntent, INPUT_FILE_REQUEST_CODE);
+        Log.d(TAG, "Started chooser");
+        return true;
+    }
+
+    /**
+     * Pass through activity result to XWalkViewInternal.
+     * See <a href="http://developer.android.com/reference/android/app/Activity.html">
+     * android.app.Activity.onActivityResult()</a>.
+     * @param requestCode passed from android.app.Activity.onActivityResult().
+     * @param resultCode passed from android.app.Activity.onActivityResult().
+     * @param data passed from android.app.Activity.onActivityResult().
```

Comment:
For the param explanation, we can follow it from [onActivityResult](https://developer.android.com/reference/android/app/Activity.html#onActivityResult(int, int, android.content.Intent))